src/main/kotlin/io/icure/md/client/apis/MedTechApi.kt

```diff
@@ -73,24 +108,53 @@ class MedTechApi(
         fun shortLivedCacheMaxSize(maxSize: Long) = apply { this.shortLivedCachesMaxSize = maxSize }
 
         fun iCureUrlPath(iCureUrlPath: String) = apply { this.iCureUrlPath = iCureUrlPath }
+        fun userName(username: String) = apply { this.userName = username }
+        fun password(password: String) = apply { this.password = password }
         fun authorization(authorization: String) = apply { this.authorization = authorization }
+        fun authServerUrl(authServerUrl: String?) = apply { this.authServerUrl = authServerUrl }
+        fun authProcessId(authProcessId: String?) = apply { this.authProcessId = authProcessId }
         fun addKeyPair(keyId: String, publicKey: RSAPublicKey, privateKey: RSAPrivateKey) =
             apply { this.rsaKeyPairs[keyId] = privateKey to publicKey }
 
+        private fun basicAuth(userName: String, password: String) =
+            "Basic ${java.util.Base64.getEncoder().encodeToString("$userName:$password".toByteArray())}"
+
         fun build(): MedTechApi {
-            if (authorization == null) {
+            if (userName == null || password == null) {
                 throw IllegalArgumentException("In order to request iCure APIs, you need to provide your credentials")
             }
 
-            if (rsaKeyPairs.isEmpty()) {
-                throw IllegalArgumentException("In order to encrypt/decrypt your data, you need to provide at least a RSA Key Pair")
-            }
-
             return MedTechApi(
-                iCureUrlPath, authorization!!, rsaKeyPairs, defaultLanguage, shortLivedCachesDuration,
-                shortLivedCachesMaxSize
+                iCureUrlPath = iCureUrlPath,
+                authorization = authorization ?: basicAuth(userName = userName!!, password = password!!),
+                rsaKeyPairs = rsaKeyPairs,
+                defaultLanguage = defaultLanguage,
+                shortLivedCachesDuration = shortLivedCachesDuration,
+                shortLivedCachesMaxSize = shortLivedCachesMaxSize,
+                authServerUrl = authServerUrl,
+                authProcessId = authProcessId
             )
         }
+
+        companion object {
+            fun from(api: MedTechApi): Builder {
+                return Builder()
```

Comment:
Complete with the other information that you have (all except the rsaKeys) 
For the userName / Password, you can instead provide the baseApis (so that you provide credentials without exposing them) and so have only one method, providing only the previous MedTechApi